query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -222,6 +222,78 @@ class SumAndMaxOverTimeFuncHD(maxColID: Int) extends ChunkedRangeFunction[Transi
   }
 }
 
+/**
+  * Sums Histograms over time and also computes Max over time of a Max field.
+  */
+class AvgWithSumAndCountOverTimeFuncD(countColId: Int) extends ChunkedRangeFunction[TransientRow] {
+  private val sumFunc = new SumOverTimeChunkedFunctionD
+  private val countFunc = new SumOverTimeChunkedFunctionD
+
+  override final def reset(): Unit = {
+    sumFunc.reset()
+    countFunc.reset()
+  }
+
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, sumFunc.sum / countFunc.sum)
+  }
+
+  import BinaryVector.BinaryVectorPtr
+
+  final def addChunks(tsVector: BinaryVectorPtr, tsReader: bv.LongVectorDataReader,
```

Comment:
I thought we were just going to use average column?   If so why is this still needed?

Reply:
AvgOverTime cannot use avg column since average of averages is not the same.